Sources/ClientGtk/ServerListView.swift

```diff
@@ -0,0 +1,93 @@
+import SwiftCrossUI
+import DeltaCore
+
+enum DetailState {
+  case addition, server(ServerDescriptor), error(String)
+}
+
+class ServerListViewState: Observable {
+  @Observed var servers: [ServerDescriptor] = []
+  @Observed var detailState = DetailState.addition
+}
+
+struct ServerListView: View {
+  var completionHandler: (ServerDescriptor) -> Void
+
+  var state = ServerListViewState()
+
+  var body: some ViewContent {
+    NavigationSplitView {
+      VStack {
+        Button("Add server") {
+          state.detailState = .addition
+        }
+
+        ScrollView {
+          ForEach(state.servers) { server in
+            Button(server.name) {
+              state.detailState = .server(server)
+            }
+            .padding(Edge.Set.bottom, 5)
+          }
+        }
+      }
+      .frame(minWidth: 100)
+      .padding(Edge.Set.trailing, 10)
+    } detail: {
+      VStack {
+        switch state.detailState {
+          case .addition:
+            additionView
+          case .server(let server):
+            Text(server.name)
+            if let port = server.port {
+              Text(server.host + ":" + String(port))
```

Comment:
In Swift it's generally better practice to use string interpolation: `"\(server.host):\(port)"` (also shorter)

Reply:
Haha I tried JS-style interpolation and it didn’t work and just gave up at that point lol, will do.

Reply:
Understandable hahah, the Swift way isn't very easy to guess (haven't seen it in any other languages)